packages/filebrowser/src/browser.ts

```diff
@@ -101,9 +102,19 @@ export class FileBrowser extends Widget {
       tooltip: this._trans.__('Refresh File List')
     });
 
+    const toggleHiddenFile = new ToolbarButton({
+      icon: circleEmptyIcon,
+      onClick: () => {
+        this._showHiddenFiles = !this._showHiddenFiles;
+        document.querySelectorAll<HTMLLIElement>("li[data-is-dot]").forEach(el => el.style.display = this._showHiddenFiles ? "flex" : "none");
+      },
```

Comment:
The file listing rendering is complex and for performance reasons reuses existing HTML nodes; this means that we cannot just toggle visibility on nodes that currently exist or we will have a repeat of https://github.com/jupyterlab/jupyterlab/issues/8860 (but for hidden files, with far worse consequences) each time we refresh the listing (enter a directory/clicking "refresh", etc). That issue was fixed in
https://github.com/jupyterlab/jupyterlab/pull/10100 and you will also find a way to persist setting in there.

In general there are two ways to approach the problem:
- do not render files starting with dot in the listing
- hide files starting with dot with CSS but keep them in the listing

The former may have a performance benefit on rendering when hidden files are not shown (less DOM nodes for the browser to handle), but will require refresh on toggle; I would encourage you to go this route.

If however you want to go the CSS route (or as a general advice for future) I would recommend implementing it by toggling a class - let's name it `jp-mod-dot-hidden` - on the listing (or `FileBrowser` node which contains the listing) and having a CSS rule like:

```csss
.jp-mod-dot-hidden li[data-is-dot] {
    display: none
}
```

This way you avoid unnecessary query and iteration on (potentially dozens or hundreds of) DOM nodes and let the browser handle it with the most optimized code for this purpose you can get. Still I recommend not showing these nodes in the first place.